apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -700,11 +700,11 @@ handle_timeout(
         Session0#{s := S, shared_sub_s := SharedSubS, stream_scheduler_s := SchedS}
     ),
     {ok, [], push_now(Session)};
-handle_timeout(_ClientInfo, ?TIMER_BUMP_LAST_ALIVE_AT, Session0 = #{s := S0}) ->
-    S = emqx_persistent_session_ds_state:commit(bump_last_alive(S0)),
+handle_timeout(_ClientInfo, ?TIMER_COMMIT, Session0 = #{s := S0}) ->
+    S = emqx_persistent_session_ds_state:commit(S0),
     Session = set_timer(
-        ?TIMER_BUMP_LAST_ALIVE_AT,
-        bump_interval(),
+        ?TIMER_COMMIT,
+        commit_interval(),
         Session0#{s := S}
     ),
```

Comment:
I think it's possible to set this timer conditionally, based on the value of `session_state.dirty` flag. E.g.

`
ensure_commit_timer(Session) ->
   ...
`

Then this function can be added to `push_now` and `pull_now` functions (and maybe some other places that touch the state).

Reply:
✅  We may also track last commit time in the session (not state) and not schedule the commit timer but simply commit if
* state is dirty
* there is no timer
* last commit was long ago enough